src/context-providers/display/types.ts

```diff
@@ -34,17 +34,15 @@ export enum EModalType {
     Custom = "custom",
 }
 
-export interface IDiscardChangesModalProps {
-    type: EModalType.DiscardChanges;
-}
-
-export interface IBulkEditModalProps {
-    type: EModalType.BulkEdit;
+export interface IModalProps {
+    type: EModalType;
+    onClickActionButton?: () => void;
+    index?: number;
 }
 
-export interface ICustomModalProps {
-    type: EModalType.Custom;
-}
+export type TDiscardChangesModalProps = IModalProps;
+export type TBulkEditModalProps = IModalProps;
+export type TCustomModalProps = IModalProps;
```

Comment:
can revert this change so we have a dedicated interface for each modal type

while `IModalProps` works, it accept any `EModalType` as the type, this may cause confusion in the future

in fact, it's best not to create logic we dont use yet, e.g. handling the bulk edit / custom modals when we are not there yet. because we dont use it, we cant assess properly whether they are developed correctly